zulip_bots/zulip_bots/lib.py

```diff
@@ -102,7 +102,74 @@ def __init__(self, name: str, email: str) -> None:
         self.email = email
         self.mention = '@**' + name + '**'
 
-class ExternalBotHandler:
+class BotHandler(Protocol):
+
+    email: str
+    full_name: str
+    storage: Any
+
+    def reset_transcript(self) -> None:
+        pass
+
+    def identity(self) -> BotIdentity:
+        pass
+
+    def react(
+        self,
+        message: Dict[str, Any],
+        emoji_name: str
+    ) -> Dict[str, Any]:
+        pass
+
+    def send_message(
+        self,
+        message: (Dict[str, Any])
+    ) -> Dict[str, Any]:
+        pass
+
+    def send_reply(
+        self,
+        message: Dict[str, Any],
+        response: str,
+        widget_content: Optional[str] = None
+    ) -> Dict[str, Any]:
+        pass
+
+    def update_message(
+        self,
+        message: Dict[str, Any]
+    ) -> Optional[Dict[str, Any]]:
+        pass
+
+    def get_config_info(
+        self,
+        bot_name: str,
+        optional: bool = False
+    ) -> Dict[str, Any]:
+        pass
+
+    def upload_file_from_path(self, file_path: str) -> Dict[str, Any]:
+        pass
+
+    def upload_file(self, file: IO[Any]) -> Dict[str, Any]:
+        pass
+
+    def unique_reply(self) -> Dict[str, Any]:
+        pass
+
+    def unique_response(self) -> Dict[str, Any]:
+        pass
+
+    def ensure_unique_response(
```

Comment:
Aren't these just test helpers? Similarly reset_transcript and maybe others?

Reply:
These are test helpers and they were added because there were a few failing tests when `bot_handler` was assigned the type `BotHandler` in the next commit.

Reply:
Will still take a look at improving it if possible.

Reply:
@neiljp yeah, I considered removing the helper functions which were not really necessary.